main_app/views.py

```diff
@@ -0,0 +1,40 @@
+from django.shortcuts import render
+from django.views import View
+from .igdb_api import IGDB
+
+
+class MainPageView(View):
```

Comment:
You're using wrong base class, please use `templateView` here.
Also do not perform post requests here